src/MongoBson.pas

```diff
@@ -1484,14 +1484,14 @@ function TBsonBuffer.finishObject: Boolean;
 begin
   child := FSubNativeBson.Pop;
   Result := bson_append_document_end(GetCurrNativeBson, child);
-  bson_destroy(child);
+  FreeMem(child);
```

Comment:
If I use bson_free() here I would have to use bson_new(), which would initialize bson, right?